amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/KinesisDataFetcher.java

```diff
@@ -120,7 +121,13 @@ public DataFetcherResult getRecords() {
 
         if (nextIterator != null) {
             try {
-                return new AdvancingResult(getRecords(nextIterator));
+                GetRecordsResponse getRecordsResponse = getRecords(nextIterator);
+                while (!isValidResponse(getRecordsResponse)) {
```

Comment:
We need some backoff strategy here, to avoid spamming the application with logs and to avoid breaching the 5 TPS limit. I would rather have this handle at the higher layer, instead of looping here. So move the `isValidResponse` check inside the `getRecords(nextIterator)` and throw `RetryableRetrievalException` when we fail the validation. The higher layer like PrefetchRecordPublisher should be taking care of the retry subjecting to the TPS limits.

Reply:
Addressed